### PR TITLE
xreftype support array

### PR DESCRIPTION
**Pull request by HanqingZheng-ms**
xreftype support array

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/docfx?pullRequest=7904) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/docfx?pullRequest=7904) before we can accept your contribution.<br/><br/>:x: HanqingZheng-ms [sign now](https://cla.dotnetfoundation.org/dotnet/docfx?pullRequest=7904)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/docfx?pullRequest=7904) it.</sub>

File: src/docfx/lib/schema/JsonSchemaTransformer.cs [hunk 1]
```diff
@@ -468,9 +468,9 @@ private string GetXrefHref(FilePath file, string uid, int uidCount, bool isRootL
 
                 errors.AddIfNotNull(xrefError);
 
-                if (xrefSpec != null && schema.XrefType != null && !schema.XrefType.Equals(xrefSpec.SchemaType, StringComparison.OrdinalIgnoreCase))
+                if (xrefSpec != null && schema.XrefType != null && !schema.XrefType.Contains(xrefSpec.SchemaType))
                 {
-                    errors.Add(Errors.Xref.XrefTypeInvalid(content, schema.XrefType, xrefSpec.SchemaType));
+                    errors.Add(Errors.Xref.XrefTypeInvalid(content, string.Join(",", schema.XrefType), xrefSpec.SchemaType));
```
Comment (yufeih) [inline src/docfx/lib/schema/JsonSchemaTransformer.cs:473]: Pass the array and use `StringUtility.Join` in Errors.cs
```suggestion
                    errors.Add(Errors.Xref.XrefTypeInvalid(content, schema.XrefType, xrefSpec.SchemaType));
```

File: docs/specs/schema.yml [hunk 2]
```diff
@@ -797,10 +802,29 @@ inputs:
         }
       }
     }
+  _themes/ContentTemplate/schemas/d_type.schema.json: |
```
Comment (yufeih) [inline docs/specs/schema.yml:805]: Add a negative case that produces a warning

**Review by yufeih — APPROVED** [review]

**Pull request merged**